### ISSUE TITLE
`asm!` xmm16-31 register usage is reported as zmm on i686 targets in diagnostic

### ISSUE THREAD

**Issue by Fulgen301** [rust-lang/rust#159409]
### Code

```Rust
pub fn foo() {
    unsafe {
        std::arch::asm!(
            "",
            lateout("xmm0") _,
            lateout("xmm8") _,
            lateout("xmm16") _,
            )
    }
}
```

### Current output

```Shell
error: cannot use register `xmm8`: register is only available on x86_64
 --> <source>:6:13
  |
6 |             lateout("xmm8") _,
  |             ^^^^^^^^^^^^^^^^^

error: cannot use register `zmm16`: register is only available on x86_64
 --> <source>:7:13
  |
7 |             lateout("xmm16") _,
  |             ^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

### Desired output

```Shell
error: cannot use register `xmm8`: register is only available on x86_64
 --> <source>:6:13
  |
6 |             lateout("xmm8") _,
  |             ^^^^^^^^^^^^^^^^^

error: cannot use register `xmm16`: register is only available on x86_64
 --> <source>:7:13
  |
7 |             lateout("xmm16") _,
  |             ^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

### Rationale and extra context

Registers xmm8-xmm31 are unavailable on 32-bit x86 targets. However, the inline assembly error message reports `xmm16` - `xmm31` as `zmm` in the diagnostics (`xmm8` - `xmm15` are reported as `xmm`).

This isn't technically wrong due to the way x86 registers are laid out, but a confusing and unnecessary difference in the diagnostic.

Reproduced with `i686-pc-windows-msvc` and `i686-unknown-linux-gnu`.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.99.0-nightly (d0babd8b6 2026-07-15)
binary: rustc
commit-hash: d0babd8b6b05ef9bb65d42f928cef4129d64cf65
commit-date: 2026-07-15
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 22.1.8
```

**Comment by workingjubilee** [comment]
> This isn't technically wrong due to the way x86 registers are laid out, but a confusing and unnecessary change in the diagnostic.

Purely pedantic note for prioritization/categorization reasons: This has been this way the entire time (i.e. it is not a regression).

**Comment by Fulgen301** [comment]
Good point, adjusted the wording. I meant "change" as in "the register name is different in the diagnostic than in the source code".

**Comment by qaijuang** [comment]
> Registers xmm8-xmm31 are unavailable on 32-bit x86 targets. However, the inline assembly error message reports `xmm16` - `xmm31` as `zmm` in the diagnostics (`xmm8` - `xmm15` are reported as `xmm`).

This is happening for both xmm16..31 and ymm16..31
